.github/workflows/checks.yml

```diff
@@ -93,9 +93,7 @@ jobs:
         uses: actions/cache@v3
         with:
           path: starknet_py/tests/e2e/mock/contracts_compiled
-          key: ${{ runner.os }}-contracts-${{ hashFiles('starknet_py/tests/e2e/mock/contracts') }}
-          restore-keys: |
-            ${{ runner.os }}-contracts-
+          key: ${{ runner.os }}-contracts-${{ hashFiles('starknet_py/tests/e2e/mock/contracts', 'poetry.lock') }}
```

Comment:
Quite clever idea, this will increase the checks runtime every re-lock but I think it should be fine.

Reply:
Right, but re-locks are not that common, especially in separate PRs 🚀